### ISSUE TITLE
JSON Download, Phonology in Dutch?

### ISSUE THREAD

**Issue by rem0g** [Signbank/Global-signbank#1212]
We would need to have JSON file in Dutch language, since we set phonology in dutch parameters. 

An other option would be to have translation file with bounded translation between the parameters.

**Comment by susanodd** [comment]
Would a parameter to indicate the language suffice? This has to work for all the (current) interface languages.

English has been used (for CSV) because some languages are not filled in and so remain English. Moreover, some languages are right to left. (The browser sometimes goes berserk. Safari adds a "ltr" or "rtl" tag to input fields internally.) The API routines will need to work for all languages once such an option is made available.

The translations of all the "interface text" is in the ".po" files. The field choice choice fields, and handshapes use "model translations".

@vanlummelhuizen is the one who initially set up the model translations.

The ".po" files need to be compiled and are not dynamic. The model translations are dynamic. Thus model translations are used.

**Comment by susanodd** [comment]
@rem0g do you want the language code in the url itself? Or as a parameter to the api called by the url? Or obtained from the request?

**Comment by susanodd** [comment]
Update. It's not going to work to let Django do this automatically. It basically ignores the NL in the request and seems to arbitrarily use or not use it based on control flow. It looks like it needs to be a parameter the the method in the Gloss model that retrieves the data.

**Comment by susanodd** [comment]
This is working now.
I did the new api functions. Not the package zip one. Do you need the zip also?
I didn't do the zip one yet because it may surprise people if it all of a sudden is translating stuff.
The zip package has existed for at least 8 years.

The only weirdness is that it is translating "Video" to (lower case) "video".

**Comment by susanodd** [comment]
This is available on signbank-dev now!

**Comment by rem0g** [comment]
Ok great, now i need instructions updated in Wikipedia to obtain the Dutch JSON. I assume this is for dictionary/pakcage?

**Comment by susanodd** [comment]
My browser did this automatically. Based on what interface language I had set in Signbank.
It gets the LANGUAGE_CODE from the request.

I'll update the Wiki. I'll ask @Woseseltops. He's going to do the API tokens.

**Comment by rem0g** [comment]
Using "dictionary/package/" there is no translation of phonology parameters and it also contains a lot of less information in comparison to english JSON (1MB versus 2 MB) on signbank-dev.

**Comment by susanodd** [comment]
I didn't add it to the package url yet. I shall do that. (Because the package function is very old, I did not know what to do with it qua translation, since they use it at MPI.)

I am familiar with the difference in size. I had the same problem. (But that was related to "being logged in" versus "being anonymous".)

I made the language code be an explicit parameter of the "model method" for Gloss. The other functions that eventually "call" the model method, retrieve the language code and pass it around. Then it gets activated at the point where the translations are done. The scope of the language code activation needs to be as small as possible.

Because it is first inside a session/Django variable that is inside all the requests, then depending on if you have multiple windows open with Signbank or are logged in on multiple computers, somehow it remembers this.

In the package function, "en" is actually set in the function. So that probably interfered with the other ones.

Sorry, this explanation is confusing me myself when I read it back. I know what I'm explaining but I can't tell whether it's to follow.

Bottom line: the package function is hard-coded to use EN. (Since always.) I will make it so it also works for NL as well.

If you get "different results" and "different amounts of data", this is related to the "control flow of other stuff".
(Django keeps session variables and remembers these also in different windows and based on previous logins. If I have a different language set at home, it will be using that at the uni.)

**Comment by susanodd** [comment]
This is available on signbank now.

**Comment by Woseseltops** [comment]
> This will adversely affect the Signbank server if used often.
> I will ask @Woseseltops about this.

Let's not fix a problem before we know it is a problem ;). If discover this affects performance, let's focus on it then
